WalletWasabi/BranchNBound/RecursiveCoinSelector.cs

```diff
@@ -0,0 +1,92 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.BranchNBound
+{
+	public class RecursiveCoinSelector : Selector
+	{
+		public Money Tolerance { get; private set; } = 0UL;
+
+		private bool _optimizedCoinsFound = false;
+
+		private List<Money>? FinalCoins { get; set; }
+
+		public bool TryBranchAndBound(List<Money> coins, Money target, ulong maxTolerance, ulong toleranceIncrement, out List<Money> finalCoins)
+		{
+			var coinsDescending = coins.OrderBy(x => x);
```

Comment:
It does looks like it, but funny enough, it actually sorts it descending.

Reply:
Not on my computer when running `NewLogicSimpleTest` test:

![Screenshot 2021-12-13 120429](https://user-images.githubusercontent.com/58662979/145801346-3b1226c9-eddb-448e-935a-e1083aa53830.png)

Reply:
Ah I see, thats true, I was misreading the Auto Watch.
Then the name `coinsDescending` is misleading, but at L31, when you create the Stack, it will be put in the right order.

![image](https://user-images.githubusercontent.com/54551879/145806283-1bd9bea5-9726-498b-a0c0-79ff309c0102.png)

Reply:
> Then the name `coinsDescending` is misleading, but at L31, when you create the Stack, it will be put in the right order.

Then it should be renamed :)